sklearn/tree/_tree.pyx

```diff
@@ -608,12 +609,12 @@ cdef class Tree:
         def __get__(self):
             return self._get_value_ndarray()[:self.node_count]
 
-    def __cinit__(self, int n_features, cnp.ndarray n_classes, int n_outputs):
+    def __cinit__(self, int n_features, SIZE_t[:] n_classes, int n_outputs):
```

Comment:
@Vincent-Maladiere This one seems a bit awkward to handle because according to the code which calls this, it should be an int_p.
https://github.com/scikit-learn/scikit-learn/blob/83a877416b15f4b5f177ada333952ad204d63986/sklearn/tree/_classes.py#L245

However due to some of the tests we can get int or long here and if I used a fused type like cython.integral it does not compile successfully.

Reply:
This one is interesting: it appears that this is a bug regarding `__init__`/`__cinit__` on the Cython side; we should be using the `integral` fusedtype here. I created [an issue in Cython](https://github.com/cython/cython/issues/5243).

I don't think there is any workaround at this time, unfortunately, so we should keep the `cnp.ndarray`.

Reply:
Sounds good. Thank you for creating the related issue!

Reply:
Perhaps you can add a note here:

XXX: convertible to integral memory view once <link to Cython issue> is fixed